model/minsky.cc

```diff
@@ -266,14 +266,12 @@ namespace minsky
     GroupPtr g(new Group);
     canvas.model->addGroup(g);
     m.populateGroup(*g);
-    // Default pasting no longer occurs as grouped items or as a group within a group. Fix for tickets 1080/1098    
-    canvas.selection.clear();    
-
+    
     // convert stock variables that aren't defined to flow variables, and other fix up multiply defined vars
     g->recursiveDo(&GroupItems::items,
                    [&](Items&, Items::iterator i) {
                      if (auto v=(*i)->variableCast())
-                       if (v->defined() || v->isStock())
+                       if ((v->defined() || v->isStock()) && !v->group.lock())
```

Comment:
How can an item in g's items vector not be owned by g? This condition should never be true.